libs/api/metadata-converter/src/lib/gn4/gn4.field.mapper.ts

```diff
@@ -135,12 +135,16 @@ export class Gn4FieldMapper {
       ...output,
       recordUpdated: toDate(selectField<string>(source, 'changeDate')),
     }),
-    publicationDateForResource: (output, source) => ({
-      ...output,
-      recordPublished: toDate(
+    publicationDateForResource: (output, source) => {
+      const datePublished = toDate(
         selectField<string>(source, 'publicationDateForResource')
-      ),
-    }),
+      )
+      return {
+        ...output,
+        recordPublished: datePublished,
```

Comment:
Am I guessing right that this is kept for backwards compatibility? I think there can theoretically be a `publicationDate` field coming from the index so I would rely on this one instead of `publicationDateForResource`, but maybe you had a good reason to do so

Reply:
Here is an update: https://github.com/geonetwork/geonetwork-ui/pull/1075/files#diff-3e4fae1b2969801149a83f9b728e055e203e1b65a4b0e51c83951edfc8104573R145
Is that what you meant?

Reply:
perfect :ok_hand: thanks!